src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -335,33 +338,32 @@ internal static bool TryParseBigInteger(string? value, NumberStyles style, Numbe
 
         internal static bool TryParseBigInteger(ReadOnlySpan<char> value, NumberStyles style, NumberFormatInfo info, out BigInteger result)
         {
-            unsafe
-            {
-                result = BigInteger.Zero;
-                ArgumentException? e;
-                if (!TryValidateParseStyleInteger(style, out e))
-                    throw e; // TryParse still throws ArgumentException on invalid NumberStyles
+            ArgumentException? e;
+            if (!TryValidateParseStyleInteger(style, out e))
+                throw e; // TryParse still throws ArgumentException on invalid NumberStyles
 
-                BigNumberBuffer bignumber = BigNumberBuffer.Create();
-                if (!FormatProvider.TryStringToBigInteger(value, style, info, bignumber.digits, out bignumber.precision, out bignumber.scale, out bignumber.sign))
-                    return false;
+            BigNumberBuffer bignumber = BigNumberBuffer.Create();
```

Comment:
I realize the casing was `bignumber` before this PR, but that's not our typical naming convention so it might be nice to fix while we're here:
```suggestion
            BigNumberBuffer bigNumber = BigNumberBuffer.Create();
```